processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -246,24 +261,26 @@ public void run() {
             } catch (RuntimeException e) {
                 log.error("Offset commit failed before closing consumer", e);
             }
-            contexts.close();
-
             processors.destroySingletonScope(scope.subscriptionId());
-
-            consumeManager.close();
-            log.info("ProcessorSubscription {} terminated in {} ms", scope,
-                     timer.elapsedMillis());
-
-            updateState(SubscriptionStateListener.State.TERMINATED);
-            metrics.close();
-            shutdownFuture.complete(null);
+            log.info("ProcessorSubscription {} terminated in {} ms", scope, timer.elapsedMillis());
```

Comment:
There are several procedures which might take certain time after this, until complete termination at this line. https://github.com/line/decaton/pull/116/files#diff-37fa89e1da134b048d335ebf4753a63374e56c59dd65882fc1e84ac1aad44cacR283

Shouldn't we log when terminated completely?

Reply:
Yeah, thought about something similar, but concluded to leave this here because of:

* This message intends to tell the time taken to terminate subscription "thread"
* likewise, I think it's strange to see "terminated ..." message if one called `.close()` on subscription w/o starting it
* because the message here depends on the timer, we have to to bring it somehow to `cleanUp` which is unnecessarily complex.

Reply:
In contrast, I put state change notification for `TERMINATED` in real-end because I thought it could be used for some purpose to trigger some processsing after exactly everything of this subscription to have destroyed, so I thought it could be bit danger to call it at the end of thread termination.

Reply:
> This message intends to tell the time taken to terminate subscription "thread"

Hmm, even "thread"'s termination not completed yet at this line I think? ("thread is terminated" == "run() returns" right?)

So better to log after `run` returns ?

> because the message here depends on the timer, we have to to bring it somehow

I see, that's right.
Hm, then let's just get rid of timer and add another log at the start of termination procedure?
We can see the time to the shutdown by log timestamp

Reply:
> Hmm, even "thread"'s termination not completed yet at this line I think? ("thread is terminated" == "run() returns" right?)

That's true, but the remaining procedure are negligible cheap and are supposed to finish very quickly isn't it?
If the purpose of this logging is to tell the exact timing of `ProcessorSubscription`'s destroy, I would agree with you, but I think the purpose of this log is to tell how long did it take to complete the cleanup sequence where the main part comes from processor's destruction.

>  Hm, then let's just get rid of timer and add another log at the start of termination procedure?

That works, but less convenient than as-is because users have to find two lines and compute duration by themselves.
I always prefer kind logging as possible as we can, so this style still fulfils the purpose IMO. 

If you think this log message gonna be confusing, shall we change it to be more precises like "ProcessorSubscription thread {} cleaned up in {} ms" ?